src/engine/index.ts

```diff
@@ -11,13 +11,31 @@ const check = (
   pageViews: PageView[],
   any = false
 ): boolean => {
-  const checkedConditions = conditions.map((condition) =>
-    evaluateCondition(condition, pageViews)
-  );
+  // Instead of going through the entire array again with an `.includes`
+  // method call, let's check _during_ other operations. This has a
+  // potential savings of `n` | n => conditions.length.
```

Comment:
I agree here, but I would recommend the same using `some`/`every` for a cleaner implementation.

```
const check = (
  conditions: EngineCondition<AudienceDefinitionFilter>[],
  pageViews: PageView[],
  any = false
): boolean => {
  return conditions[any ? 'some' : 'every']((condition) =>
    evaluateCondition(condition, pageViews)
  );
};
```

Reply:
Cool - sounds good to me. I'll refactor as such. Thanks!